doc-checking-app-api/src/main/java/uk/gov/di/authentication/app/lambda/DocAppCallbackHandler.java

```diff
@@ -287,4 +289,55 @@ private APIGatewayProxyResponseEvent redirectToFrontendErrorPage() {
+            throws ParseException, DocAppCallbackException {
+        LOG.info(
+                "Attempting to generate error response using state. CustomDocAppClaimEnabled: {}",
+                configurationService.isCustomDocAppClaimEnabled());
+        if (isAccessDeniedErrorAndStatePresent(queryStringParameters)) {
+            LOG.info("access_denied error and state param are both present");
+            var clientSessionId =
+                    authorisationService
+                            .getClientSessionIdFromState(
+                                    State.parse(queryStringParameters.get("state")))
+                            .orElseThrow(
+                                    () ->
+                                            new DocAppCallbackException(
+                                                    "ClientSessionId could not be found using state param"));
+            LOG.info("ClientSessionID found using state");
+            attachLogFieldToLogs(CLIENT_SESSION_ID, clientSessionId);
+            attachLogFieldToLogs(GOVUK_SIGNIN_JOURNEY_ID, clientSessionId);
+            var clientSession =
+                    clientSessionService
+                            .getClientSession(clientSessionId)
+                            .orElseThrow(
+                                    () ->
+                                            new DocAppCallbackException(
+                                                    "No client session found with given client sessionId"));
+            LOG.info("ClientSession found using clientSessionId");
+            var authenticationRequest =
+                    AuthenticationRequest.parse(clientSession.getAuthRequestParams());
+            var errorDescription =
+                    Optional.ofNullable(queryStringParameters.get("error_description"))
+                            .orElse(OAuth2Error.ACCESS_DENIED.getDescription());
+            var errorObject = new ErrorObject(queryStringParameters.get("error"), errorDescription);
+            LOG.info(
+                    "ErrorObject created for session cookie not present. Generating error response back to RP");
+            return generateAuthenticationErrorResponse(authenticationRequest, errorObject);
+        } else {
+            LOG.warn(
+                    "Session Cookie not present and access_denied or state param missing from error response");
+            throw new DocAppCallbackException(
+                    "Session Cookie not present and access_denied or state param missing from error response");
+        }
+    }
+
+    private boolean isAccessDeniedErrorAndStatePresent(Map<String, String> queryStringParameters) {
+        return configurationService.isCustomDocAppClaimEnabled()
+                && queryStringParameters.containsKey("error")
+                && queryStringParameters.get("error").equals(OAuth2Error.ACCESS_DENIED.getCode())
+                && queryStringParameters.containsKey("state");
```

Comment:
A very minor thing - I reckon we should put another condition on the end of this chain which is "state param is non-empty".

I _think_ that it's possible (since the callback happens in browser and urls can be changed) that it could be an empty string (`&state=&...`) which we would want to fail on?

Not a blocker though, if we roll this into a future PR.

Reply:
👍 Good point. Agreed to resolve in a future PR.